api/processors/actions/student/accepted_offered_group.py

```diff
@@ -0,0 +1,37 @@
+from django.db import transaction
+
+from api.models import Coordinator, Group, Student
+from api.models.choices.log_event_type import CoordinatorLogEventType, StudentLogEventType
+from api.models.choices.status import StudentProjectStatus
+from api.models.log_event import CoordinatorLogEvent, StudentLogEvent
+from api.processors.actions.student import StudentActionProcessor
+
+
+class StudentAcceptedOfferedGroupProcessor(StudentActionProcessor):
+    """Student accepted offer to join group"""
+
+    def __init__(self, student: Student, coordinator: Coordinator, group: Group):
+        self.coordinator = coordinator
+        self.group = group
+        super().__init__(student)
+
+    @transaction.atomic
+    def process(self) -> None:
+        self._create_log_events()
+        self._set_statuses()
+
+    def _set_statuses(self) -> None:
+        self.student.project_status = StudentProjectStatus.STUDYING
+        self.student.situational_status = ""
+        self.student.status_since = self.timestamp
+        self.student.save()
+
+    def _create_log_events(self) -> None:
+        StudentLogEvent.objects.create(
+            student=self.student, to_group=self.group, type=StudentLogEventType.STUDY_START
```

Comment:
Please, double check that, we also have event `accepted_offer`

Reply:
Yes, I know, but thanks for checking. I think this event will be used with the automatic communication with students when a new group is created. Maybe at some point we will introduce the functionality when a coordinator clicks something like "propose to join existing group" and the bot contacts the student. Then this `accepted_offer` event may be needed. For now, though, the situation is that the group is already studying and the coordinator contacts another student and asks them: do you want to join group such-and-such? If the student says yes, the coordinator adds them to the group and the student is already considered "studying". Hence `STUDY_START`. Of course, we could create two events simultaneously, but I don't see any value in that.